gnd/src/main/java/com/google/android/gnd/ui/basemapselector/BasemapSelectorViewModel.java

```diff
@@ -12,8 +28,108 @@
  * com.google.android.gnd.workers.FileDownloadWorker}.
  */
 public class BasemapSelectorViewModel extends ViewModel {
+  private static final String TAG = BasemapSelectorViewModel.class.getName();
+
+  private final Flowable<ImmutableSet<Tile>> tilesStream;
+  private final LiveData<ImmutableSet<Tile>> tiles;
+  private HashSet<String> extentsPendingDownload = new HashSet<>();
+  private PublishSubject<HashSet<String>> extentsPendingDownloadStream = PublishSubject.create();
+  private HashSet<String> extentsPendingRemoval = new HashSet<>();
+  private PublishSubject<HashSet<String>> extentsPendingRemovalStream = PublishSubject.create();
+  private LiveData<String> downloadedExtents;
+  private PublishSubject<String> downloadedExtentsSubject = PublishSubject.create();
```

Comment:
Same comment here. I would expect the view model to expose all extents as a single stream.

Reply:
Done!